src/vanilla/TransformerTS.cs

```diff
@@ -22,15 +22,48 @@ public override CodeModelTS TransformCodeModel(CodeModel cm)
 
             // we're guaranteed to be in our language-specific context here.
             SwaggerExtensions.NormalizeClientModel(codeModel);
+            TransformHeaderCollectionPropertyTypes(codeModel);
             PopulateAdditionalProperties(codeModel);
             NormalizeOdataFilterParameter(codeModel);
             PerformParameterMapping(codeModel);
             CreateModelTypeForOptionalClientProperties(codeModel);
             CreateModelTypesForOptionalMethodParameters(codeModel);
             AddEnumTypesToCodeModel(codeModel);
+
             return codeModel;
         }
 
+        public void TransformHeaderCollectionPropertyTypes(CodeModelTS cm)
+        {
+            foreach (var modelType in cm.HeaderTypes)
```

Comment:
I found this transformation was only necessary for the method parameters, not the response headers, so I may delete this. It's strange because I don't remember having to do something like this in Java for method parameters. I tried updating the modeler again but it doesn't really seem to have an effect.

Reply:
It seems like for whatever reason the AutoRest modeler interprets header parameters with the HeaderCollectionPrefix extension as being exactly the type indicated in the Swagger--not a dictionary whose values are of the indicated type. Since these header parameters can also have constraints applied to them, we have to do some contortions to make it so the constraints are applied to the dictionary values, not just the keys.